kyo-zio/shared/src/main/scala/kyo/ZIOs.scala

```diff
@@ -1,50 +1,37 @@
 package kyo
 
-import ZIOs.GetZIO
 import kyo.kernel.*
+import kyo.scheduler.IOPromise
+import scala.concurrent.ExecutionContext
 import scala.util.control.NonFatal
 import zio.Exit
+import zio.Runtime
+import zio.Unsafe
 import zio.ZIO
 
-/** Effect to integrate ZIO with Kyo */
-opaque type ZIOs <: Async = GetZIO & Async
-
 object ZIOs:
 
-    /** Executes a ZIO effect and returns its result within the Kyo effect system.
-      *
-      * @param v
-      *   The ZIO effect to execute
-      * @param ev
-      *   Evidence that E is a subtype of Nothing
-      * @param tag
-      *   Tag for E
-      * @param frame
-      *   Implicit Frame
-      * @tparam E
-      *   The error type (must be a subtype of Nothing)
-      * @tparam A
-      *   The result type
-      * @return
-      *   A Kyo effect that will produce A or abort with E
-      */
-    def get[E >: Nothing: Tag, A](v: ZIO[Any, E, A])(using Frame): A < (Abort[E] & ZIOs) =
-        val task = v.fold(Result.fail, Result.success)
-        ArrowEffect.suspendMap(Tag[GetZIO], task)(Abort.get(_))
-
-    /** Executes a ZIO effect that cannot fail and returns its result within the Kyo effect system.
+    /** Lifts a zio.ZIO into a Kyo effect.
       *
-      * @param v
-      *   The ZIO effect to execute
-      * @param frame
-      *   Implicit Frame
-      * @tparam A
-      *   The result type
+      * @param zio
+      *   The zio.ZIO to lift
       * @return
-      *   A Kyo effect that will produce A
+      *   A Kyo effect that, when run, will execute the zio.ZIO
       */
-    def get[A](v: ZIO[Any, Nothing, A])(using Frame): A < ZIOs =
-        ArrowEffect.suspend(Tag[GetZIO], v)
+    def get[E, A](v: ZIO[Any, E, A])(using Frame): A < (Abort[E] & Async) =
+        IO {
+            val p = new IOPromise[E, A]
+            val result =
+                Unsafe.unsafe { implicit unsafe =>
+                    Runtime.default.unsafe.runToFuture(v.either)
```

Comment:
You can use `.exit` if you want to retain more information, ie `Die` vs `Fail`. I will admit `Exit` + `Cause` contains a significant number of possible cases.

Reply:
The `Exit` API confusing to me since methods end up producing `ZIO`. I've added tests to check the translation of ZIO failures to Kyo and it seems correct. ZIO's `die` gets propagated as a failed future.